web/modules/custom/migration/src/Plugin/migrate/process/CustomLinkPlugin.php

```diff
@@ -14,20 +14,29 @@
  * )
  */
 class CustomLinkPlugin extends ProcessPluginBase {
-
   /**
    * {@inheritdoc}
    */
-  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
-    $callback = function ($matches) {
-      $url = $matches[2];
-      $modified_url = str_replace('ehistory.osu.edu', '', $url);
-      return $matches[1] . $modified_url . $matches[3];
-    };
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property): array
+  {
+    if (!empty($value['value'])) {
+      $dom = new \DOMDocument();
+      $dom->loadHTML($value['value']);
+      $anchors = $dom->getElementsByTagName('a');
+      foreach ($anchors as $anchor) {
+        $href = $anchor->getAttribute('href');
+        $href = str_replace('http://ehistory.osu.edu', '', $href);
+        $is_valid = \Drupal::pathValidator()->isValid($href);
```

Comment:
Didn't use DI here cause faced an error, was not able to fix it fast and didn't want to spend too much time fixing it. So left it as it is.

Error message: TypeError: Drupal\migration\Plugin\migrate\process\CustomLinkPlugin::__construct(): Argument #4 ($pathValidator) must be of type Drupal\Core\Path\PathValidatorInterface, Drupal\node\Plugin\migrate\D7NodeTranslation given